browser/src/RegionalGenomicConstraintTrack.tsx

```diff
@@ -0,0 +1,296 @@
+        <rect x={60} y={0} width={25} height={10} stroke="#000" fill="#bdc9e1" />
+        <rect x={85} y={0} width={25} height={10} stroke="#000" fill="#e2e2e2" />
+        <rect x={110} y={0} width={25} height={10} stroke="#000" fill="#ffeda0" />
+        <rect x={135} y={0} width={25} height={10} stroke="#000" fill="#feb24c" />
+        <rect x={160} y={0} width={25} height={10} stroke="#000" fill="#f03b20" />
+
+        <text x={35} y={10} fontSize="10" dy="1.2em" textAnchor="middle">
+          -2.5
+        </text>
+        <text x={60} y={10} fontSize="10" dy="1.2em" textAnchor="middle">
+          -1.2
+        </text>
+        <text x={85} y={10} fontSize="10" dy="1.2em" textAnchor="middle">
+          -0.6
+        </text>
+        <text x={110} y={10} fontSize="10" dy="1.2em" textAnchor="middle">
+          0.6
+        </text>
+        <text x={135} y={10} fontSize="10" dy="1.2em" textAnchor="middle">
+          1.2
+        </text>
+        <text x={160} y={10} fontSize="10" dy="1.2em" textAnchor="middle">
+          2.5
+        </text>
+      </svg>
+    </LegendWrapper>
+  )
+}
+
+const renderNumber = (number: any) =>
+  number === undefined || number === null ? '-' : number.toPrecision(4)
+
+type RegionTooltipProps = {
+  region: {
+    z: number
+    obs_exp: number
+  }
+}
+
+const RegionTooltip = ({ region }: RegionTooltipProps) => (
+  <RegionAttributeList>
+    <div>
+      <dt>Z:</dt>
+      <dd>{renderNumber(region.z)}</dd>
+    </div>
+    <div>
+      <dt>O/E:</dt>
+      <dd>{renderNumber(region.obs_exp)}</dd>
+    </div>
+  </RegionAttributeList>
+)
+
+const SidePanel = styled.div`
+  display: flex;
+  align-items: center;
+  height: 100%;
+`
+
+const TopPanel = styled.div`
+  display: flex;
+  justify-content: flex-end;
+  width: 100%;
+  margin-bottom: 5px;
+`
+
+type OwnRegionalConstraintTrackProps = {
+  height?: number
+  start: number
+  stop: number
+  regions: {
+    start: number
+    stop: number
+    z: number
+  }[]
+}
+
+// @ts-expect-error TS(2456) FIXME: Type alias 'RegionalConstraintTrackProps' circular... Remove this comment to see the full error message
+type RegionalConstraintTrackProps = OwnRegionalConstraintTrackProps &
+  typeof RegionalGenomicConstraintTrack.defaultProps
+
+// @ts-expect-error TS(7022) FIXME: 'RegionalConstraintTrack' implicitly has type 'any... Remove this comment to see the full error message
+const RegionalGenomicConstraintTrack = ({
+  height,
+  start,
+  stop,
+  regions,
+}: RegionalConstraintTrackProps) => {
+  if (stop - start > 150000) {
+    return (
+      <Wrapper>
+        <Track
+          renderLeftPanel={() => (
+            <SidePanel>
+              <span>Regional genomic constraint (1kb scale)</span>
+              {/* TODO:(rgrant) link to blog / help text when that is ready */}
+              {/* <InfoButton topic="regional-constraint" /> */}
+            </SidePanel>
+          )}
+        >
+          {({ scalePosition, width }: any) => (
```

Comment:
We should have more specific types here too. I assume `width` is a number, and from context it looks like `scalePosition` is a function of some signature.